src/Integrators.jl

```diff
@@ -83,7 +208,19 @@ function step!(sys::System{0}, integrator::Langevin)
     set_energy_grad_dipoles!(∇E, s₁, sys)
     @. s = s + 0.5 * Δt * (f₁ + rhs_dipole(s₁, -∇E, λ)) + 0.5 * √Δt * (r₁ + rhs_dipole(s₁, ξ))
     @. s = normalize_dipole(s, sys.κs)
-    nothing
+
+    # Collect statistics about the magnitude of the drift term squared. For the
+    # energy-conserving part, it is important to use |∇E|² rather than |s×∇E|²,
+    # because the former gives direct information about the frequency of
+    # oscillation, while the latter may artificially vanish (consider, e.g., a
+    # spin nearly aligned with an external field).
+    for i in eachsite(sys)
+        s0 = sys.κs[i] # == norm(s[i])
+        ∇E² = ∇E[i]' * ∇E[i]
+        accum!(integrator.drift_squared, (1 + (s0*λ)^2) * ∇E²)
+    end
```

Comment:
Have you benched the effect of this? Perhaps it should be optional? (I imagine the cost is much less than the calculation of the gradient, but I'm curious about the rough order of magnitude.)